src/FeedRegistration.php

```diff
@@ -112,37 +112,128 @@ public function handle_feed_registration() {
 	 * Also if a different feed is registered, it will update using the URL in the
 	 * $feed_args.
 	 *
-	 * @return boolean|string
+	 * @return boolean
 	 *
 	 * @throws Exception PHP Exception.
 	 */
 	private static function register_feed() {
 
-		// Get merchant object.
-		$merchant   = Merchants::get_merchant();
-		$registered = false;
+		$merchant_id = self::check_merchant_approval_status();
+
+		if ( ! $merchant_id ) {
+			return false;
+		}
+
+		$feed_id = self::get_locally_stored_registered_feed_id();
+
+		// If the feed is not registered, try to match it to a registered feed.
+		if ( ! $feed_id ) {
+			$feed_id = Feeds::match_local_feed_configuration_to_registered_feeds( $merchant_id );
+		}
```

Comment:
If I have a locally registered feed option, then it will never search for a locale different feed and update it unless the `feed_registered` option is emptied.

My suggestion is let it search for a matching feed and create a new one or reactivate an old one if any.

```suggestion
		$feed_id = Feeds::match_local_feed_configuration_to_registered_feeds( $merchant_id );
```

Also there is a problem if new locale does not match any of Pinterest locales, but I think it is fixed with another PR. So for example if I set my locale to New Zealand then it does not update and just quits with an Exception (the exception has no meaningful message and just says something like Unsupported parameter). I assume this is a locale problem since I haven't found New Zealand from the list of supported locales by Pinterest and other locations like Australia, United Kingdom or US those do work well. I'll check this case when reviewing locale mapping PR once again.

Reply:
> My suggestion is let it search for a matching feed and create a new one or reactivate an old one if any.

This is how it was actually done before. I think that I have tried to bee a bit too smart. Great catch.

> Also there is a problem if new locale does not match any of Pinterest locales, but I think it is fixed with another PR.
Yes this is handled by https://github.com/woocommerce/pinterest-for-woocommerce/pull/687 so we don't have to think about this here. But good point nevertheless.